src/ast/nodes.ts

```diff
@@ -614,6 +616,13 @@ export interface ESLintLegacySpreadProperty extends HasLocation, HasParent {
     argument: ESLintExpression
 }
 
+/**
+ * Top-level type annotation from `typescript-eslint-parser` and `babel-eslint`
+ */
+export interface TypeAnnotation extends HasLocation, HasParent {
+    type: "TSTypeAnnotation" | "TypeAnnotation"
```

Comment:
I know there are many more possible types, but they are rather children of these top level types. And what we need here at this moment is just the top level type annotation on `Identifier`. Unless there will be introduced another eslint rule that will respect further cases of typeAnnotation or returnType. What do you think?